streams/src/main/java/org/apache/kafka/streams/kstream/MaterializedKV.java

```diff
@@ -0,0 +1,415 @@
+
+                    @Override
+                    public boolean persistent() {
+                        return bytesWindowStore.persistent();
+                    }
+
+                    @Override
+                    public void init(final ProcessorContext context, final StateStore root) {
+                        bytesWindowStore.init(context, root);
+                    }
+
+                    @Override
+                    public void flush() {
+                        bytesWindowStore.flush();
+                    }
+
+                    @Override
+                    public void close() {
+                        bytesWindowStore.close();
+                    }
+
+                    @Override
+                    public boolean isOpen() {
+                        return bytesWindowStore.isOpen();
+                    }
+
+                    @Override
+                    public void put(final Bytes key, final byte[] value) {
+                        final Windowed<Bytes> windowedBytes = windowedSerde.deserializer().deserialize(null, key.get());
+                        bytesWindowStore.put(windowedBytes.key(), value, windowedBytes.window().start());
+                    }
+
+                    @Override
+                    public byte[] putIfAbsent(final Bytes key, final byte[] value) {
+                        final Windowed<Bytes> windowedBytes = windowedSerde.deserializer().deserialize(null, key.get());
+                        final byte[] fetched = bytesWindowStore.fetch(windowedBytes.key(), windowedBytes.window().start());
+                        if (fetched != null) {
+                            return fetched;
+                        } else {
+                            bytesWindowStore.put(windowedBytes.key(), value, windowedBytes.window().start());
+                            return value;
+                        }
+                    }
+
+                    @Override
+                    public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
+                        for (final KeyValue<Bytes, byte[]> entry : entries) {
+                            put(entry.key, entry.value);
+                        }
+                    }
+
+                    @Override
+                    public byte[] delete(final Bytes key) {
+                        return putIfAbsent(key, null);
+                    }
+
+                    @Override
+                    public byte[] get(final Bytes key) {
+                        final Windowed<Bytes> windowedBytes = windowedSerde.deserializer().deserialize(null, key.get());
+                        return bytesWindowStore.fetch(windowedBytes.key(), windowedBytes.window().start());
+                    }
+
+                    @Override
+                    public KeyValueIterator<Bytes, byte[]> range(final Bytes from, final Bytes to) {
+                        final Windowed<Bytes> windowedBytesFrom = windowedSerde.deserializer().deserialize(null, from.get());
```

Comment:
Should we just reject its usage by a runtime exception?

Reply:
It depends. It's perfectly well defined, so I don't see why we'd reject the usage. But then again, this API call is only to support IQ, and as you point out later, we should register this store as a WindowStore for the purpose of IQ.